Samples for Extending Pentaho/Reference Implementations/Security/SAML 2.0/pentaho-saml/src/main/java/org/pentaho/platform/spring/security/saml/PentahoSamlUserDetailsService.java

```diff
@@ -25,13 +27,14 @@
 
   private static final String PROVIDER_NAME = "providerName";
 
-  UserDetailsService userDetailsService;
-  String selectedAuthorizationProvider;
+  private UserDetailsService userDetailsService;
+  private String selectedAuthorizationProvider;
 
   // if the selectedAuthorizationProvider is our own ( i.e. samlId ), we'll directly use our own
-  UserDetailsService samlUserDetailsService;
-  String samlId;
-
+  private PentahoSamlNativeUserDetailsService samlUserDetailsService;
```

Comment:
@pedrofvteixeira  - In this case the PentahoSamlUserDetailsService delegates to either the native SAML UserDetailsService, or one selected through the system.  There are two variables here...

private UserDetailsService userDetailsService;
private PentahoSamlNativeUserDetailsService samlUserDetailsService;

The SAML specific one gets instantiated through the beans in blueprint.xml. Then this UserDetailsService delegates to the appropriate one based on the authorization.provider specified as the constructor arg.